tests/test_image.py

```diff
@@ -194,21 +197,64 @@ def test_bad_image_type(self):
+        for color in allColorFormats:
             with self.subTest(colorFormat=color):
-                with pytest.raises(LunaSDKException) as exceptionInfo:
-                    VLImage(body=SINGLE_CHANNEL_IMAGE, imgFormat=color).save(f"test_jpeg.jpg")
-                self.assertLunaVlError(exceptionInfo, LunaVLError.InvalidBitmap)
+                if color not in failColorFormats:
+                    VLImage(body=IMAGE).save(IMG_PATH, colorFormat=color)
+                else:
+                    with pytest.raises(LunaSDKException) as exceptionInfo:
+                        VLImage(body=IMAGE).save(IMG_PATH, colorFormat=color)
+                    self.assertLunaVlError(exceptionInfo, LunaVLError.InvalidBitmap)
 
-    def test_zero_numpy_array_image(self):
+    def test_zero_numpy_array(self):
         """
-        Test image validation with a zero array
+        Test image validation with a zero array:
+            (0, 0, 0), (0, 0, 0), (0, 0, 0)
+            (0, 0, 0), (0, 0, 0), (0, 0, 0)
+            (0, 0, 0), (0, 0, 0), (0, 0, 0)
         """
-        zeroArray = np.zeros(shape=(2, 2))
-        blackImage = VLImage(body=zeroArray, filename="array")
+        zeroArray = np.zeros(shape=(3, 3, 3))
+        blackImage = VLImage.fromNumpyArray(arr=zeroArray, inputColorFormat="RGB", filename="array")
         assert blackImage.isValid()
         self.checkRectAttr(blackImage.rect)
+
+    def test_convert(self):
+        """
+        Test convert image combinations (every to every).
+        """
+        allImages = self.getColorToImageMap()
+        unsupportedConversions = (
+            (ColorFormat.IR_X8X8X8, ColorFormat.R16),
+            (ColorFormat.R8, ColorFormat.R16),
+            (ColorFormat.B8G8R8, ColorFormat.R16),
+            (ColorFormat.R8G8B8, ColorFormat.R16),
+            (ColorFormat.B8G8R8X8, ColorFormat.R16),
+            (ColorFormat.R8G8B8X8, ColorFormat.R16),
+            (ColorFormat.R16, ColorFormat.IR_X8X8X8),
+            (ColorFormat.R16, ColorFormat.R8),
+            (ColorFormat.R16, ColorFormat.B8G8R8),
+            (ColorFormat.R16, ColorFormat.R8G8B8),
+            (ColorFormat.R16, ColorFormat.B8G8R8X8),
+            (ColorFormat.R16, ColorFormat.R8G8B8X8),
+        )
+
+        for source in set(allImages):
+            sourceImage = allImages[source]
+            for target in set(allImages):
+                with self.subTest(source=source.name, target=target.name):
+                    if (source, target) in unsupportedConversions:
+                        with pytest.raises(LunaSDKException) as exceptionInfo:
+                            sourceImage.convert(target)
+                        self.assertLunaVlError(
+                            exceptionInfo, LunaVLError.InvalidConversion.format("Requiered conversion not implemented")
+                        )
+                    else:
+                        sourceImage.convert(target)
```

Comment:
+ tests for init image from  differents formats with validation

Reply:
test added